tests/Doctrine/Tests/DoctrineTestCase.php

```diff
@@ -5,10 +5,51 @@
 namespace Doctrine\Tests;
 
 use PHPUnit\Framework\TestCase;
+use PHPUnit\Runner\Version;
 
 /**
  * Base testcase class for all Doctrine testcases.
  */
 abstract class DoctrineTestCase extends TestCase
 {
+    public static function assertDoesNotMatchRegularExpression(
+        string $pattern,
+        string $string,
+        string $message = ''
+    ): void {
+        // Forward-compatibility wrapper for phpunit 9 : can be removed once phpunit 8 / php 7.2 support is dropped.
+        if (self::isPhpunit9()) {
+            parent::assertDoesNotMatchRegularExpression($pattern, $string, $message);
+        } else {
+            self::assertNotRegExp($pattern, $string, $message);
+        }
```

Comment:
Let's do feature detection instead?


```suggestion
    $methodName = method_exists(TestCase::class, 'assertDoesNotMatchRegularExpression')?
        'assertDoesNotMatchRegularExpression',
        'assertNotRegExp';
        
    return parent::$methodName($pattern, $string, $message);
```